src/routerd_lib/handlers/proxy.cpp

```diff
@@ -45,26 +46,82 @@ namespace NAC {
         return hosts.front();
     }
 
+#ifdef AC_DEBUG_ROUTERD_PROXY
+    void TRouterDProxyHandler::print_outgoing_request(std::shared_ptr<TRouterDRequest> request) const{
+        auto &outgoing_request = request->GetOutGoingRequest();
+        std::cerr << "== OUTGOING REQUEST == " << std::endl;
+        std::cerr << "=== headers ===" << std::endl;
+        for (auto && [header, values] : outgoing_request.Headers()) {
+            std::cerr << "  " << header << ":";
+            if(values.size() == 1) {
+                std::cerr << " " << values.at(0) << std::endl;
+            } else if (values.size() >= 2 ) {
+                for (auto && value : values) {
+                    std::cerr << std::endl << "  " << value << std::endl;
+                }
+            } else {
+                std::cerr << " (empty)" << std::endl;
+            }
+        }
+        std::cerr << "=== end of headers ===" << std::endl;
+        std::cerr << "=== parts ===" << std::endl;
+        for (auto &&part : outgoing_request.Parts()) {
```

Comment:
`for (auto &&part` -> `for (auto&& part`

And empty line before "for", of course.